### ISSUE TITLE
Use intersection of SNPs between GWAS sumstats and target genotypes

### ISSUE THREAD

**Issue by eugenniee** [opain/GenoPred#130]
From https://opain.github.io/GenoPred/pipeline_technical.html#Workflow my understanding is that sumstat_prep step is done independnetly from format_target step, thus internal score files (produced by format_target) may have SNPs that are absent in target genotypes. Would it be recomended to first constrain my GWAS sumstats to the set of SNPs available in the genotypes, so that e.g. P+T method selects the SNPs that are available?

**Comment by opain** [comment]
You are correct. The pipeline does not consider the target sample when deriving the score files for polygenic scoring. The advantage of this is that you don't need to create a score file for each target sample. The pipeline avoids the issue of missing SNPs by restricting analyses to variants that are typically available after imputation (HapMap3 SNPs), and throwing an error if the target sample does not contain enough of the reference SNPs (<70%). This approach is pragmatic and will help to make the variance explained more generalisable, since other datasets will also likely have some missing HapMap3 variants. The reference-standardisation process is particularly important when using polygenic scores for prediction modelling.

However, as you highlight, the limitation of this approach is that some key variants may not be present in your target sample, which will impact the variance explained by the score. If you are not using the polygenic scores for prediction modelling, but are instead using them for inference (like estimating genetic overlap), then restricting to variants present in the target sample might offer an advantage.

My recommendation would be to not adjust the GWAS sumstats for a given target sample, unless your target sample is missing a lot of reference (HapMap3) variants. But if you want to restrict the analysis to reference variants present in your target sample, then subsetting the GWAS sumstats would achieve this. I will consider adding an option for the pipeline that restricts analyses to the target sample variants. Alternatively, you could create a reference for the pipeline that is tailored for your target sample, by restricting it to variants present in your target sample. There is some documentation on using your own reference here: https://opain.github.io/GenoPred/pipeline_readme.html#Using_your_own_reference
